storage: deflake TestReservationQueue

BramGruneir opened the pull request:
There was a small race between setting the reservationTimeout and the bookie
starting and using the default value.  When the default value is used, it sets
a 2 minute timeout which causes the test to fail as the loop in bookie.start
won't make any forward progress until after that timeout.

Fixes #7373

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7493)

<!-- Reviewable:end -->

BramGruneir committed:
storage: deflake TestReservationQueue

There was a small race between setting the reservationTimeout and the bookie
starting and using the default value.  When the default value is used, it sets
a 2 minute timeout which causes the test to fail as the loop in bookie.start
won't make any forward progress until after that timeout.

Fixes #7373

tamird commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/reservation.go, line 100 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7493#-KLIWvSWMREbxuSZ1QN1:-KLIWvSWMREbxuSZ1QN2:1455492883) ([raw file](https://github.com/cockroachdb/cockroach/blob/cb8f4876576a4fdf49cbd1aac2875dd4f9686c41/storage/reservation.go#L100)):*
> ```Go
> 		metrics:            metrics,
> 		reservationTimeout: reservationTimeout,
> 		maxReservations:    envutil.EnvOrDefaultInt("max_reservations", defaultMaxReservations),
> ```

ISTM that these parameters want the same treatment. Perhaps you could extract them into a bookieConfig struct to avoid growing the parameter list.

---

*[storage/reservation_test.go, line 253 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7493#-KLIXTsqlyVYT_NS7rFw:-KLIXTsqlyVYT_NS7rFx:-801013583) ([raw file](https://github.com/cockroachdb/cockroach/blob/cb8f4876576a4fdf49cbd1aac2875dd4f9686c41/storage/reservation_test.go#L253)):*
> ```Go
> 
> // expireNextReservation advances the manual clock to one nanosecond passed the
> // next expiring reservation and waits until exactly one reservation has expired.
> ```

changes from here down look random, but ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7493)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/reservation.go, line 100 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7493#-KLIWvSWMREbxuSZ1QN1:-KLIXkAFqxZK7KUhI83n:1899336662) ([raw file](https://github.com/cockroachdb/cockroach/blob/cb8f4876576a4fdf49cbd1aac2875dd4f9686c41/storage/reservation.go#L100)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ISTM that these parameters want the same treatment. Perhaps you could extract them into a bookieConfig struct to avoid growing the parameter list.
</details>
The other parameters are not used in the main loop.  So the current treatment is fine.  If I do have to go back in for more edits, I'll do just that.

---

*[storage/reservation_test.go, line 253 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7493#-KLIXTsqlyVYT_NS7rFw:-KLIY3g_hB-9EtVWM63_:1962187574) ([raw file](https://github.com/cockroachdb/cockroach/blob/cb8f4876576a4fdf49cbd1aac2875dd4f9686c41/storage/reservation_test.go#L253)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> changes from here down look random, but ok
</details>
I just cleaned up the function to expire only a single reservation.  When I first wrote this function, there was a need to expire multiple reservations at at time.  This is no longer the case and I just removed some of old code that wasn't useful.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7493)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7493#-:-KLIZtsQpVjoPyrkgU4d:1417088721)*
<!-- Sent from Reviewable.io -->
